src/main/kotlin/com/kotlindiscord/bot/extensions/CleanExtension.kt

```diff
@@ -0,0 +1,162 @@
+                        """
+                        Cleaning with :
+                        Users: ${user?.joinToString(", ") { "${it.username}#${it.discriminator}" }}
+                        Regex: ${regex?.joinToString(", ")}
+                        Channels: ${`in`?.joinToString(", ") {
+                            it.id.longValue.toString()
+                        } ?: message.channelId.longValue}
+                        Since: ${since?.id?.longValue}
+                        Bot-only: $botonly
+                        Count: $count
+                        Force: $force
+                        """.trimIndent()
+
+                    val channels =
+                    if (since != null) {
+                        if (!`in`.isNullOrEmpty()) {
+                            message.channel.createMessage(":x: Cannot use the `in` and `channel` options together")
+                            return@action
+                        }
+                        listOf(since.channelId.longValue)
+                    } else if (`in`.isNullOrEmpty()) {
+                        listOf(message.channelId.longValue)
+                    } else {
+                        `in`.map { it.id.longValue }
+                    }
+
+                    val userIds = user?.map { it.id.longValue }
+                    val sinceTimestamp = since?.timestamp?.minusMillis(SINCE_OFFSET)
+
+                    logger.debug { cleanNotice }
+
+                    for (channelId in channels) {
+                        var query = bot.kord.cache.query<MessageData> {
+                            MessageData::channelId eq channelId
+
+                            if (!userIds.isNullOrEmpty()) run {
+                                MessageData::authorId `in` userIds
+                            }
+
+                            if (botonly) run {
+                                MessageData::authorIsBot eq true
+                            }
+
+                            if (!regex.isNullOrEmpty()) run {
+                                MessageData::content predicate { regex.all { regex -> regex.matches(it) } }
+                            }
+
+                            if (sinceTimestamp != null) run {
+                                MessageData::timestamp predicate {
+                                    Instant.parse(it).isAfter(sinceTimestamp)
+                                }
+                            }
+                        }.toCollection()
+
+                        if (count > 0) {
+                            query = query.sortedBy { Instant.parse(it.timestamp) }.reversed().slice(0..count)
+                        }
+
+                        if (query.size > MAX_DELETION_SIZE) {
+                            if (
+                                message.getAuthorAsMember()?.roles?.toList()
+                                    ?.contains(config.getRole(Roles.ADMIN)) == false
+                                // Basically, if the author isn't an admin.
```

Comment:
If you feel like it needs a comment, and it's this messy, then it's probably better to extract it to a function.